corehq/apps/dump_reload/sql/load.py

```diff
@@ -96,10 +99,44 @@ def terminate_workers(dbalias_to_workerqueue):
             loaded_model_counts.update(model_labels)
         return loaded_model_counts
 
-    def line_to_object(self, line):
+    def max_pk_in_source(self, model_label, dump_meta):
+        from corehq.apps.dump_reload.sql.dump import SqlDataDumper
+        return dump_meta[SqlDataDumper.max_pk_slug][model_label]
+
+    @memoized
+    def max_pk_in_target(self, model_label):
+        model_class = get_model_class(model_label)
+        return model_class.objects.aggregate(Max('pk'))['pk__max']
+
+    def offset_pk(self, obj, dump_meta):
+        from corehq.apps.dump_reload.sql.dump import is_pk_type_int
+        model_class = get_model_class(obj['model'])
+        if not is_pk_type_int(model_class):
+            return
+
+        def offset(val, model_label):
+            return val + max(
+                self.max_pk_in_source(model_label, dump_meta),
+                self.max_pk_in_target(model_label),
```

Comment:
Why do you want the max of source / target? I would have thought you'd always want the max target.